pkg/manifests/config.go

```diff
@@ -83,15 +83,16 @@ type PrometheusOperatorConfig struct {
 }
 
 type PrometheusK8sConfig struct {
-	LogLevel            string                               `json:"logLevel"`
-	Retention           string                               `json:"retention"`
-	NodeSelector        map[string]string                    `json:"nodeSelector"`
-	Tolerations         []v1.Toleration                      `json:"tolerations"`
-	Resources           *v1.ResourceRequirements             `json:"resources"`
-	ExternalLabels      map[string]string                    `json:"externalLabels"`
-	VolumeClaimTemplate *monv1.EmbeddedPersistentVolumeClaim `json:"volumeClaimTemplate"`
-	RemoteWrite         []monv1.RemoteWriteSpec              `json:"remoteWrite"`
-	TelemetryMatches    []string                             `json:"-"`
+	LogLevel                      string                               `json:"logLevel"`
+	Retention                     string                               `json:"retention"`
+	NodeSelector                  map[string]string                    `json:"nodeSelector"`
+	Tolerations                   []v1.Toleration                      `json:"tolerations"`
+	Resources                     *v1.ResourceRequirements             `json:"resources"`
+	ExternalLabels                map[string]string                    `json:"externalLabels"`
+	VolumeClaimTemplate           *monv1.EmbeddedPersistentVolumeClaim `json:"volumeClaimTemplate"`
+	RemoteWrite                   []monv1.RemoteWriteSpec              `json:"remoteWrite"`
+	TelemetryMatches              []string                             `json:"-"`
+	AdditionalAlertManagerConfigs *v1.SecretKeySelector                `json:"additionalAlertManagerConfigs"`
```

Comment:
Thinking more about the user experience here, I'm not sure that the configuration should just be a reference to an uncontrolled secret. The risk is high that the provided configuration is invalid and breaks Prometheus. Also not all fields of the [`alertmanager_config`](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#alertmanager_config) section are worth exposing (for instance configuring an external Alertmanager endpoint only needs the `static_configs` service discovery) and some might be not be usable in practice (for instance any field like `ca_file` that reference a file on disk).

It might be more appropriate to expose the main fields that are required to configure external Alertmanager endpoints. From there, CMO could generate the secret to be referenced in the Prometheus resource.

Here is the rough idea:

```go
type PrometheusK8sConfig struct {
    ...
    AlertmanagerConfigs []AlertmanagerConfig `json:"additionalAlertManagerConfigs"`
}

type AlertmanagerConfig struct {
    # URL to the Alertmanager server.
    URL net.URL `json:"url"`
    Timeout string `json:"timeout"`
    # Reference to a key in a Secret containing the certificate authority to verify the server's certificate. 
    CA *v1.SecretKeySelector
    # Authn parameters (to be defined)
    ...
}
```